### PR TITLE
Allow manual stock adjustment on order items

### LINKED ISSUE #21754: Need option to manually reduce stock on order level
zd-1530164

In WooCommerce 3.4.7, the ability to manually reduce stock within the order was available:

Image Link: https://cld.wthms.co/5t4nKR

When following the above process, and saving the order (keeping order in Payment Pending Status), stock is reduced (per order notes, outside of the screenshot).

That feature was removed in WooCommerce 3.5 and order stock is automatically updated only after the order is set to processing (after payment received).

While auto-updating is good and saves a step, this takes the ability to change stock on the order level out of the hands of store owners.

In this case, a store owner needs to make sure products do not go out of stock while waiting for payment (which, happens after items are received in this case), but could happen often in Wholesale arrangements where more time is needed than is covered in hold stock options.

### PR DESCRIPTION
**Pull request by rrennick**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Enable selecting order item rows - #21837 
- Add manual item stock adjustment in order edit

Closes #21754 .

### How to test the changes in this Pull Request:

1. Rebuild admin.css `sass assets/css/admin.scss assets/css/admin.css`.
2. Enable the manual stock adjustment buttons `add_filter( 'woocommerce_order_allow_manual_stock_adjustment', '__return_true' );`.
3. Create a new order with at least 2 products and 2 quantities.
4. Click on a product line item on the order. 
- It should highlight.
- The `Reduce Stock` and `Increase Stock` buttons should slide down.
5. Click on it again.
- It should unhighlight.
6. Click on a product line item on the order again.
7. Click the edit pencil.
- The `Reduce Stock` and `Increase Stock` buttons should slide up.
8. Click `Cancel` or `Save`.
9. Click on a product line item on the order again.
10. Click `Reduce Stock`.
- On refresh an order note should be added indicating the product stock was reduced by the line item quantity.
11. On the same item, increase the quantity of the item and save.
12. Reduce the stock on that item again. 
- On refresh an order note should be added indicating the product stock was reduced by the net change in the line item quantity.
13. Increase the stock on that item.
- Order note indicating the full quantity reduced should be added back to stock.
14. Reduce stock on all product line items. Order notes should reflect those changes.
15. Increase stock should also calculate net adjustments and limit stock increases to the amount reduced.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

Additional note/question for reviewer: In restoring the `tbody tr.selected` CSS, I did not restore the td pointer in https://github.com/woocommerce/woocommerce/blob/3.4.7/assets/css/admin.scss#L1454:L1456. If that should be restored as well, let me know.

### Changelog entry

> Add manual item stock adjustment in order edit

### CODE REVIEW

File: includes/class-wc-ajax.php [hunk 1]
```diff
@@ -1910,6 +1910,138 @@ public static function save_variations() {
...
+					continue;
+				}
+
+				$_product = $order_item->get_product();
+				if ( $_product && $_product->exists() && $_product->managing_stock() ) {
+					$net_stock_change = $order_item->get_quantity();
+					$reduced = $order_item->get_meta( '_reduced_stock', true );
+					if ( ! $reduced ) {
+						$reduced = $net_stock_change;
+					} elseif ( $net_stock_change <= 								$reduced ) {
```
Comment (claudiulodro) [inline includes/class-wc-ajax.php]: Why so much whitespace?

**Comment by claudiulodro** [comment]
Thanks for your work on reimplementing this feature. The admin UI part of everything looks good in my testing.

I wasn't able to get the increase/reduce stock buttons to do anything, though. Looking at the code in this PR there doesn't seem to be anything hooked to the buttons that will call the ajax methods?

Also requesting a review from @mikejolley since he will be back by the time 3.5.3 is released.

File: includes/class-wc-ajax.php (hunk 1 above)
Comment (rrennick) [reply includes/class-wc-ajax.php]: Thanks, I'm not sure why phpcs didn't catch that one.

**Comment by rrennick** [comment]
> Looking at the code in this PR there doesn't seem to be anything hooked to the buttons that will call the ajax methods?

This has been added.

**Pull request closed without merging**